### ISSUE TITLE
Add index column to table builder.

### ISSUE THREAD

**Issue by mkinternet** [rainlab/builder-plugin#296]
I table builder i miss feature to add index column. There could be column with index and dropdown list for each created field in table containing options:
- primary key $table->primary('id');
- unique key $table->unique('field');
- index key $table->index('field');

**Comment by bennothommo** [comment]
@mkinternet I think an "Indexes" tab that allows the control of indexes would be a good idea. Indexes can be compound (ie. contain more than one column in the index) and as you pointed out, can be different types of index (ie. primary, unique, fulltext), so it probably should belong to its own tab.

**Comment by LukeTowers** [comment]
I'm not sure this is worth the effort to make a GUI for given that you can always manually edit the migration in the code editor before it is run to add whatever indexes you want.

**Comment by bennothommo** [comment]
True, but I reckon it'd be very appealing to be able to sort out the indexes and the schema in one swoop. I'm keen to give this a try down the line.

**Comment by mkinternet** [comment]
OMG i did't realised that popup window with migration file is editable :D. I think you can place hint in this popup window containing text "You can edit migration file before saving it"

**Comment by bennothommo** [comment]
@mkinternet Happy to accept a PR with that hint added if you're willing to submit one. :)

**Comment by daftspunk** [comment]
> OMG i did't realised that popup window with migration file is editable :D

Big reveal :grin: October has many nice things like this. Documenting them all can make the system too clunky and spoil some of the magic. Once the user learns about it, he/she doesn't need to see it every time.
